### ISSUE TITLE
Modify here you can disconnect the user mikrotik ros

### ISSUE THREAD

**Issue by CN-SaNDuo** [lirantal/daloradius#31]
Daloradius + mikrotik ros can not disconnect online users! I modified the source code as follows:
 rep-online.php
In line 241 add: Framed-IP-Address = $ ip

`<a class='toolTip' href='config-maint-disconnect-user.php?username=$username&nasaddr=$nasip&customattributes=Acct-Session-Id=$acctsessionid,Framed-IP-Address=$ip'>".
`
Test command：echo User-Name=777,Framed-IP-Address="192.168.1.20" | radclient -x -F 192.168.2.39:3799 disconnect testing123
Test video：http://pan.baidu.com/s/1c2tcHTQ 
blog ：http://blog.csdn.net/chinasanduo/article/details/62231831#t15
QQ：1064331064
Sorry for my bad english

**Comment by CN-SaNDuo** [comment]
![Uploading QQ截图20170423032735.jpg…]()

**Comment by CN-SaNDuo** [comment]
Example Disconnect-Request
FreeRADIUS server (radiusd) supports sending Disconnect-Request via the update COA and update disconnect Unlang statements. You can also send disconnect packets to a Disconnect enabled NAS with radclient as follows:

Translation of the original  ; Sample Disconnect-Request
The FreeRADIUS server (radiusd) support by updating the COA to send Disconnect-Request, disconnect and update the Unlang statement. You can also use radclient to disconnect the data packet to the disconnected NAS, as shown below:

# echo "Acct-Session-Id=D91FE8E51802097" > packet.txt
# echo "User-Name=somebody" >> packet.txt
# echo "NAS-IP-Address=10.0.0.1" >> packet.txt

# cat packet.txt | radclient -x 10.0.0.1:3799 disconnect ''secret''

Sending Disconnect-Request of id 214 to 10.0.0.1 port 3799
      Acct-Session-Id = "D91FE8E51802097"
      User-Name = "somebody"
      NAS-IP-Address = 10.0.0.1
rad_recv: Disconnect-ACK packet from host 10.0.0.1 port 3799, id=214, length=20
Note: The actual attributes which need to be sent in the Disconnect-Request and the port you send the packet to may vary depending on your brand of NAS and it&#39;s configuration. Though the RFC states the destination UDP port should be 3799 for Disconnect-Requests , Cisco brand equipment uses the non standard UDP port 1700 by default for POD.

For Mikrotik try:

Translation of the original "" "note: the actual needs of property and send data in the Disconnect-Request package in the port may be due to the NAS brand and its configuration. Although the RFC statement UDP port for the disconnect request should be 3799, but the Cisco brand equipment default to use non standard UDP port 1700 as POD.

To try Mikrotik:
 # cat packet.txt radclient -r 10.0.0.1:1700 disconnect''secret''1 |
The address of a document (Freeradius) Http://wiki.freeradius.org/protocol/Disconnect-Messages

I went to the Mikrotik forum to find it on CoA and PoD
The address of a document (Mikrotik) Http://wiki.freeradius.org/protocol/Disconnect-Messages
The document that does not support PoD
Note: RouterOS does not support POD (open group) another RADIUS access request packet performs a similar function as Disconnect Messages
When I TM B, people is how to play? Look at the code that would be no problem, I will not do why!
ROS log   hotspot, error, info, debug    Radius disconnect with no IP provided-- with no IP provided and the --Radius
At last I found in a daloradius
echo "User-Name='777'," | radclient -c '1' -n '3' -r '3' -t '3' -x '192.168.2.39:3799' 'disconnect' 'testing123' 2>&1
But not so well! Simply kicking off the user. Then what do we do? I think of a change in the code that haha

echo User-Name=777,Framed-IP-Address="40.0.0.254" | radclient -d /opt/freeradius-server-2.1.10/raddb -x -F 192.168.2.39:3799 coa testing123
Ver-2.1.10/raddb -x -F 192.168.2.39:3799 COA testing123
Sending CoA-Request of ID 148 to 192.168.2.39 port 3799
User-Name = "777"
Framed-IP-Address = 40.0.0.254
rad_recv: CoA-ACK packet from host 192.168.2.39 port 3799, id=148, length=36
144: -
NAS-Identifier = "MikroTik"
NAS-IP-Address = 192.168.2.39
This is not good
In this change
echo User-Name=777,Framed-IP-Address="40.0.0.254" | radclient -d /opt/freeradius-server-2.1.10/raddb -x -F 192.168.2.39:3799 disconnect testing123
Ha ha a success
User-Name=777, Then the user account
Framed-IP-Address="40.0.0.254"This is the user to obtain the ip address
I changed COA This made it disconnect
 
echo User-Name=777,Framed-IP-Address="40.0.0.254" | radclient -x -F 192.168.2.39:3799 disconnect testing123
This is good for
The command that also modify the daloradius rep-online.php
Add in the 241 line:Framed-IP-Address=$ip

`<a Class= toolTip; href='config-maint-disconnect-user.php?username=$username&nasaddr=$nasip&customattributes=Acct-Session-Id=$acctsessionid,Framed-IP-Address=$ip' >`


The test demo video daloradius
Link: Http://pan.baidu.com/s/1c2tcHTQ Password: 6wiy

**Comment by lirantal** [comment]
Hi @CN-SaNDuo 

Thanks for sending an update on this.
Can you summarize better what was the issue and what was the fix? seems like it was related to how dalo disconnects but you posted so much text on the last comment that I don't understand what it relates to.

**Comment by lirantal** [comment]
@CN-SaNDuo nevermind - I skimmed through your issue and understand it.
Thanks a bunch!

I'll commit this fix soon to master branch and ping you.

**Comment by lirantal** [comment]
thanks @CN-SaNDuo, it's now fixed: https://github.com/lirantal/daloradius/commit/b3e2524220a34e6021a3d106a8c85f8171a354b4
